File: cpp/3487.cpp
#include <vector>
#include <unordered_set>
#include <algorithm>
#include <numeric>
using namespace std;


class Solution {
public:
    int maxSum(vector<int>& nums) {
        if (nums.size() == 1) return nums[0];
        unordered_set<int> seen;
        int maxN = nums[0];

        for (int n : nums) {
            maxN = max(maxN, n);
            if (n > 0) {
                seen.insert(n);
            }
        }
        seen.insert(maxN);
        return accumulate(seen.begin(), seen.end(), 0);
    }
};

File: py/3487.py
class Solution:
    def maxSum(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        
        maxNum = nums[0]
        seen = set()
        for n in nums:
            maxNum = max(maxNum, n)
            if n > 0 and n not in seen:
                seen.add(n)
        seen.add(maxNum)
        
        return sum(seen)
